### PR TITLE
8336257: Additional tests in jmxremote/startstop to match on PID not app name

### PR DESCRIPTION
**Pull request by kevinjwalls**
Follow-on from JDK-8207908.

Two tests are broken by that change:
sun/management/jmxremote/startstop/JMXStartStopTest.java	
sun/management/jmxremote/startstop/JMXStatusPerfCountersTest.java

These are additional tests which use jcmd and an application name pattern to find a process.
They should use the PID to avoid finding a process from some other concurrent test invocation.
So it's good to have the same kind of change applied here too.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8336257](https://bugs.openjdk.org/browse/JDK-8336257): Additional tests in jmxremote/startstop to match on PID not app name (**Bug** - P2)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20138/head:pull/20138` \
`$ git checkout pull/20138`

Update a local copy of the PR: \
`$ git checkout pull/20138` \
`$ git pull https://git.openjdk.org/jdk.git pull/20138/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20138`

View PR using the GUI difftool: \
`$ git pr show -t 20138`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20138.diff">https://git.openjdk.org/jdk/pull/20138.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20138#issuecomment-2223173622)

### CODE REVIEW

File: test/jdk/sun/management/jmxremote/startstop/JMXStartStopTest.java [hunk 1]
```diff
@@ -347,6 +347,8 @@ public synchronized void start() throws InterruptedException, IOException, Timeo
                                                 "the requested port not being available");
                     }
                     pid = p.pid();
+                    jcmd = new ManagementAgentJcmd(p, verbose);
+
```
Comment (plummercj) [inline test/jdk/sun/management/jmxremote/startstop/JMXStartStopTest.java:351]: I don't think you want a blank line here.

Comment (kevinjwalls) [reply test/jdk/sun/management/jmxremote/startstop/JMXStartStopTest.java]: I don't mind the additional space, but ok have removed.

**Review by plummercj — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]

**Review by alexmenkov — APPROVED** [review]

**Comment by dcubed-ojdk** [comment]
@kevinjwalls - When do you expect to integrate this fix? It looks like
we're seeing 34 failures per Tier3 job set due to this issue.

**Review by dcubed-ojdk — APPROVED** [review]
The fix looks good to me. I presume it has been tested
with a Mach5 Tier3 job set.

**Comment by kevinjwalls** [comment]
Yes, tier3 test job came back ok, will integrate now.
Thanks for the reviews!

**Pull request closed without merging**